azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -351,6 +351,22 @@ def _reset_agent_cgroup_setup(self):
             except Exception as err:
                 logger.warn("Error while resetting the quotas: {0}".format(err))
 
+        def _is_enforcement_supported(self, controller):
```

Comment:
What do you think about moving this to the controller implementations? 
CpuControllerV1 -> return True
CpuControllerV2 -> return False
_MemoryController -> return False

Then during cgroup initialization, we can just check controller.is_enforcement_supported()

Reply:
Good idea, but for extensions we setup the limits before we start the extension, and we will not have controllers objects to get this information

Reply:
Each controller implementation has a static can_enforce() check:
CpuControllerV1 -> return True
CpuControllerV2 -> return False
_MemoryController -> return False

Cgroupapi implementations have a can_enforce_cpu() check:
CgroupApiV1 -> return CgroupV1.CPU_CONTROLLER in self._cgroup_mountpoints and CpuControllerV1.can_enforce() 
CgroupApiV2 -> return CgropuV2.CPU_CONTROLLER in self._controllers_enabled_at_root and CpuControllerV2.can_enforce()

Then cgroupconfigurator just calls self._cgroupapi.can_enforce_cpu() 


^ This allows us to avoid these if/else conditions, but I'm also ok with leaving as is if this overcomplicates

Reply:
I think we can even simply without static can_enforce(), not sure if we need this

Cgroupapi implementations have a can_enforce_cpu() check:
CgroupApiV1 -> return CgroupV1.CPU_CONTROLLER in self._cgroup_mountpoints
CgroupApiV2 -> return False

can_enforce_memory() -> return False

Reply:
Sounds good to me